### ISSUE TITLE
Unable to access `build_trains` from Spaceship

### ISSUE THREAD

**Issue by raxityo** [fastlane/fastlane#10912]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

```
$ bundle exec fastlane spaceship
Logging into to iTunes Connect (***@***.com)...
Successfully logged in to iTunes Connect

Logging into the Developer Portal (***@***.com)...
Successfully logged in to the Developer Portal

---------------------------------------
| Welcome to the spaceship playground |
---------------------------------------

Enter docs to open up the documentation
Enter exit to exit the spaceship playground
Enter _ to access the return value of the last executed command

Just enter the commands and confirm with Enter
[1] pry(#<Spaceship::Playground>)> Spaceship::Tunes::Application.all.first.build_trains
Spaceship::Client::UnexpectedResponse: Temporary iTunes Connect error: <?xml version="1.0" encoding="UTF-8"?>
<String>There is no route for the path 'apps/MY_APP_ID_HERE/trains/'.</String>
from /Users/rax/.rvm/gems/ruby-2.2.3/gems/fastlane-2.64.1/spaceship/lib/spaceship/tunes/tunes_client.rb:799:in `rescue in build_trains'
[2] pry(#<Spaceship::Playground>)>
```

Note: In the error log, it shows actual app ID instead of `MY_APP_ID_HERE`.

It appears that the URL to get `build_trains` now includes platform in it, which looks like: `apps/APP_ID/platform/[ios|appletvos]/trains`.

**Comment by mrand01** [comment]
I'm experiencing the same thing with the `latest_testflight_build_number` fastlane action, which appears to depend on this property existing as well.

**Comment by jhandguy** [comment]
What exactly are you experiencing @mrand01? To me the `latest_testflight_build_number` cannot find the build number even when specifying the wanted version...

**Comment by runloop** [comment]
I'm getting 'There is no route for the path' when using Pilot as well.

**Comment by chrusart** [comment]
Me too

**Comment by pj4533** [comment]
co-sign, just got this (after the 503 from yesterday lifted!)

**Comment by chrusart** [comment]
@raxityo , I changed the source and:
```<String>There is no route for the path 'apps/APP_ID/platform/ios/trains/'.</String>```

**Comment by ohayon** [comment]
👋  Hey everyone - thanks for raising this up. Looks like Apple made some changes to TestFlight to support different builds being distributed to different groups (which is sweet), but in the process they broke our APIs. We are starting to dig into this now and will keep this thread updated with progress. 🚀

**Comment by ohayon** [comment]
I have a PR that is a work in progress here if you'd like to follow along: https://github.com/fastlane/fastlane/pull/10921

**Comment by raxityo** [comment]
Thanks, @ohayon.

Am I correct to assume that these API changes are only impacting TestFlight related actions such as getting build train info, version history etc.

Or there's also an impact on actions that simply upload builds to iTunesConnect?

**Comment by ohayon** [comment]
Hey @raxityo - as far as we can tell uploading should work fine, but let us know if you run into any issues with that! 🚀

**Comment by ohayon** [comment]
Hey everyone - we have made some changes on a branch for this PR https://github.com/fastlane/fastlane/pull/10921. We would love to get some feedback on it to see if it fixes issues for you folks. Would it be possible for some of you to pull down this branch and run your fastlane execution against this branch and then report back? Thanks!! 🐙

**Comment by ebmifa** [comment]
So, what should be the new end point for trains? .../ra/apps/XXXXXXX/platform/ios/trains does not seem to work.

**Comment by taquitos** [comment]
@ebmifa https://itunesconnect.apple.com/testflight/v2/providers/<team_id>/apps/<app_id>/platforms/ios/trains

**Comment by ebmifa** [comment]
Thanks @taquitos, I assume that the <team_id> is what's listed under the membership in Apple's Developer Portal?

**Comment by taquitos** [comment]
I don't think that's right ☹️ I'm pretty sure that's the ID of the ID 🙄

**Comment by ebmifa** [comment]
"ID of the ID"?

**Comment by taquitos** [comment]
Yeah, I'm still not clear on the difference between `team_id` and `provider_id` We seem to use them interchangeably. I think the `team_id` you're thinking about looks like `123ZXY456`
Whereas, the `provider` in that url is actually an integer, but it seems to be an ID of my team->
```
},
	"provider": {
		"providerId": <number>,
		"name": "<team name>",
		"contentTypes": ["SOFTWARE"]
	},
```

**Comment by ebmifa** [comment]
Ok, found a way to find the provider id: https://searchr.uservoice.com/knowledgebase/articles/775704-i-m-an-app-developer-how-do-i-find-my-provider-id

**Comment by sam9046** [comment]
Looks like iTunes Connect are still making changes, our builds went ahead fine last night (after failing the night before with the wrong build number), but they had a really weird build number (went from 7 to 428)...

Just wanted to raise that looks like they are still making changes....

**Comment by sharat** [comment]
I also faced similar error and here's the build output

```
➜ bundle exec fastlane pilot builds
[19:14:21]: To not be asked about this value, you can specify it using 'username'
[19:14:21]: Your Apple ID Username: my@email.com
[19:14:32]: Login to iTunes Connect (my@email.com)
[19:14:37]: Login successful

Looking for related GitHub issues on fastlane/fastlane...

Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
bundler: failed to load command: fastlane (/Users/user/.rbenv/versions/2.4.1/bin/fastlane)
Spaceship::Client::UnexpectedResponse: [!] Temporary iTunes Connect error: <?xml version="1.0" encoding="UTF-8"?>
<String>There is no route for the path 'apps/1284759091/trains/'.</String>

  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/spaceship/lib/spaceship/tunes/tunes_client.rb:799:in `rescue in build_trains'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/spaceship/lib/spaceship/tunes/tunes_client.rb:775:in `build_trains'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/spaceship/lib/spaceship/tunes/build_train.rb:48:in `all'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/spaceship/lib/spaceship/tunes/application.rb:264:in `build_trains'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/spaceship/lib/spaceship/tunes/application.rb:302:in `all_processing_builds'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/pilot/lib/pilot/build_manager.rb:94:in `list'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/pilot/lib/pilot/commands_generator.rb:84:in `block (2 levels) in run'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:66:in `run!'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/pilot/lib/pilot/commands_generator.rb:159:in `run'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/pilot/lib/pilot/commands_generator.rb:12:in `start'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/fastlane/lib/fastlane/cli_tools_distributor.rb:58:in `take_off'
  /Users/user/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.64.1/bin/fastlane:20:in `<top (required)>'
  /Users/user/.rbenv/versions/2.4.1/bin/fastlane:22:in `load'
  /Users/user/.rbenv/versions/2.4.1/bin/fastlane:22:in `<top (required)>'
```

**Comment by ohayon** [comment]
Hey @sharat - did you get that error when you ran against this branch?

**Comment by sharat** [comment]
@ohayon Which branch?

I saw your PR to [update TestFlight's new endpoints](https://github.com/fastlane/fastlane/pull/10921). I think it should do the work.

**Comment by mrand01** [comment]
Any updates on this?  I'm trying out 2.65.0 right now, but am still experiencing the same problem - iTunes Connect always returns "1" as the latest version.

**Comment by taquitos** [comment]
Still on this @mrand01, we're hoping to have something for ya in the next day or so. These endpoint changes were big.

**Comment by sharat** [comment]
@taquitos it’s fixed in 2.66 isn’t it?
